### ISSUE TITLE
ValueError: 'vertices' must be a 2D list or array with shape Nx2

### ISSUE THREAD

**Issue by flaviodoliveira** [matplotlib/basemap#512]
I'm using Python 3.8.5
Matplotlib version 3.1.2
and basemap-1.2.2+dev

I'm running something like:
```

    llon = lons.min()
    llat = lats.min()
    ulon = lons.max()
    ulat = lats.max()

    fig, ax = plt.subplots()
    
    m = Basemap(llcrnrlon=llon, llcrnrlat=llat, urcrnrlon=ulon,
                urcrnrlat=ulat, resolution='h')

    m.drawcoastlines(linewidth=0.8)
    m.drawcountries(linewidth=0.8)
    m.drawstates(linewidth=0.8)


```
and then I got this:

>   File "plt_RAIN.py", line 62, in <module>
>     m.drawcoastlines(linewidth=0.8)
>   File "/usr/local/lib/python3.8/dist-packages/mpl_toolkits/basemap/__init__.py", line 1859, in drawcoastlines
>     coastlines = LineCollection(self.coastsegs,antialiaseds=(antialiased,))
>   File "/usr/lib/python3/dist-packages/matplotlib/collections.py", line 1333, in __init__
>     self.set_segments(segments)
>   File "/usr/lib/python3/dist-packages/matplotlib/collections.py", line 1348, in set_segments
>     self._paths = [mpath.Path(_seg) for _seg in _segments]
>   File "/usr/lib/python3/dist-packages/matplotlib/collections.py", line 1348, in <listcomp>
>     self._paths = [mpath.Path(_seg) for _seg in _segments]
>   File "/usr/lib/python3/dist-packages/matplotlib/path.py", line 129, in __init__
>     raise ValueError(
> ValueError: 'vertices' must be a 2D list or array with shape Nx2
> 

I checked everything, but I'm used to run this on basemap 1.0.7 and everything was ok.
And when I run without m.drawcoastlines(), there is no problem.

What can I do to run it?

**Comment by guziy** [comment]
I have a similar problem. I think this might be because your domain does not intersect any coastlines? I have this reproducer:

```python
# fails 
from mpl_toolkits.basemap import Basemap
b = Basemap(projection="cea", llcrnrlon=267.5, urcrnrlon=283.995, llcrnrlat=41.35, urcrnrlat=51, resolution="l") 
b.drawcoastlines()
```

*Note*: The above fails in basemap 1.2.1 and works in basemap 1.1.0, regression somewhere...

But if I extend a bit to have an intersection with the ocean coastlines (in this case to the Hudson Bay) it works

```python
# works 
from mpl_toolkits.basemap import Basemap
b = Basemap(projection="cea", llcrnrlon=267.5, urcrnrlon=283.995, llcrnrlat=41.35, urcrnrlat=51.8, resolution="l") 
b.drawcoastlines()
```


@jswhit I know we are not supposed to bug you with this project, but maybe you have a workaround for this...

**Comment by molinav** [comment]
@guziy With the latest `basemap` I am not able to reproduce it. If I type your first snippet in IPython, I get this:
```
(py38) vic@onyx:~$ python -m IPython --pylab=Tk
Python 3.8.12 (default, Jan  3 2022, 10:33:21)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.30.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from mpl_toolkits.basemap import Basemap
   ...: import matplotlib.pyplot as plt
   ...: b = Basemap(projection="cea",
   ...:             llcrnrlon=267.5, urcrnrlon=283.995,
   ...:             llcrnrlat=41.35, urcrnrlat=51, resolution="l")
   ...: b.drawcoastlines()
   ...: plt.show()

In [2]: from mpl_toolkits import basemap

In [3]: basemap.__version__
Out[3]: '1.3.2'
```
![plot_first_snippet](https://user-images.githubusercontent.com/9979942/161107549-458ea990-445f-4bc8-b26a-5afb91dde5ec.png)

For this example I am using the `basemap` wheel in PyPI, which bundles a relatively old GEOS library (3.5.1). Could it be that your `basemap` installation uses the GEOS from your system, which is probably newer, and the new GEOS does not get on well with the current implementation?

**Comment by guziy** [comment]
@molinav Could you please try to create the following conda environment:

```bash
conda create -n py39 python=3.9 basemap ipython
conda activate py39
```
I am not sure how to check geos version...

**Comment by molinav** [comment]
@guziy The `basemap` conda package needs an update at some point, but I haven't still had time to learn a bit on how to refactor the `basemap` conda recipe appropriately after my split of `basemap` into three parts.

From other issue (https://github.com/matplotlib/basemap/issues/522#issuecomment-1021245967) it seems that we are blocked at the moment to GEOS < 3.9.

**Comment by guziy** [comment]
* I don't know about conda recipies, but willing to learn, are they in this repo?
* @flaviodoliveira could you please check if the latest develop branch fixes the issue for you as well?

**Comment by molinav** [comment]
The conda recipe is available in a separate repo:
https://github.com/conda-forge/basemap-feedstock

There were a couple of external attempts to update the recipe but they did not succeed or were closed before finishing:
https://github.com/conda-forge/basemap-feedstock/pull/92
https://github.com/conda-forge/basemap-feedstock/pull/93

Unfortunately, I could not still find time to learn about conda recipes to provide my own update there.

**Comment by molinav** [comment]
@guziy This issue and your fix were included in `basemap` 1.3.3. Since April, a bit of development on the `conda-forge` side has been ongoing, and we now have up-to-date `basemap` packages in `conda-forge` as well as in PyPI.

The latest `basemap` release (1.3.5) has been deployed to PyPI and `conda-forge` today, it also includes this bugfix of course, and it solves the problem about linking with newer GEOS versions (#522). So I think we can safely close this issue!
